sql-db/narayana-transactions/src/test/java/io/quarkus/ts/transactions/TransactionCommons.java

```diff
@@ -119,14 +130,28 @@ public void verifyRollbackForNarayanaProgrammaticApproach() {
                 .statusCode(HttpStatus.SC_NO_CONTENT);
     }
 
+    @Order(5)
     @Tag("QUARKUS-2492")
     @Test
     public void smokeTestNarayanaProgrammaticTransactionTrace() {
         String operationName = "GET /transfer/accounts/{account_id}";
         given().get("/transfer/accounts/" + ACCOUNT_NUMBER_LUIS).then().statusCode(HttpStatus.SC_OK);
-        verifyRestRequestTraces(operationName);
+        verifyRequestTraces(operationName);
+    }
+
+    @Order(6)
+    @Test
+    public void verifyJdbcTraces() {
+        for (String operationName : getExpectedJdbcOperationNames()) {
+            verifyRequestTraces(operationName);
+        }
     }
 
+    protected String[] getExpectedJdbcOperationNames() {
+        return new String[] { "SELECT mydb.account", "INSERT mydb.journal", "UPDATE mydb.account" };
```

Comment:
Do you know what makes the operation names different for Oracle and MS SQL databases?

`AgroalOpenTelemetryWrapper` is used to get the details, so there are some specifics coming from the drivers or directly from the DB engine?

Reply:
> Do you know what makes the operation names different for Oracle and MS SQL databases?

MS SQL is different due to DB name used by `SqlServerService`
https://github.com/quarkus-qe/quarkus-test-framework/blob/04f65af3d42547cb2e756ef9a509581af2b56474/quarkus-test-service-database/src/main/java/io/quarkus/test/bootstrap/SqlServerService.java#L7

Oracle database - TBH I don't know why it doesn't have table name in that trace as I didn't think it should concern us. Due to how this is implemented, there is no way this is Quarkus bug, either it is specific of Oracle driver, Agroal impl. or OpenTelemetry bug. I can dig into it if you want, but only on demand :-)

> AgroalOpenTelemetryWrapper is used to get the details, so there are some specifics coming from the drivers or directly from the DB engine?

`AgroalOpenTelemetryWrapper` is a trick (out of my head, please don't take this word for word): `OpenTelemetry` provides `java.sql.Datasource` wrapper that basically spy for every traced SQL/opened connected and whatever else can be tracked, but OTEL is optional dep in Agroal so we can't refer it from files that can be used without OTEL dep, so there is wrapper. Wrapper provides Agroal DS wraper that partially delegates to OTEL wrapper. But Agroal specifics still requires using Agroal DS too (=we can't wrap Agroal DS with OTEL wrapper)

Reply:
> I can dig into it if you want, but only on demand :-)

+1, please check that. It looks suspicious.